partner_delivery_link/models/res_partner.py

```diff
@@ -0,0 +1,34 @@
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    delivery_order_ids = fields.Many2many(
+        "stock.picking", compute="_compute_delivery_count"
+    )
+    delivery_count = fields.Integer(compute="_compute_delivery_count")
+
+    def _compute_delivery_count(self):
+        for partner in self:
+            delivery_order_ids = self.env["stock.picking"].search(
+                [
+                    ("partner_id", "=", self.id),
+                    ("picking_type_id.code", "=", "outgoing"),
+                ]
+            )
+            partner.delivery_order_ids = delivery_order_ids
+            partner.delivery_count = len(delivery_order_ids)
```

Comment:
Can you use the function of this picking search? So, It can make picking search flexible.